core/execution/engine_checkpoint.go

```diff
@@ -34,6 +34,10 @@ func (e *Engine) Checkpoint() ([]byte, error) {
 }
 
 func (e *Engine) Load(ctx context.Context, data []byte) error {
+	if len(data) == 0 {
```

Comment:
I don't think this is necessary since in the checkpoint engine we only call `Load()` on an engine if data exists:
```
cpData := cp.Get(k)
if len(cpData) == 0 {
    continue
}
```
and the map is already set to non-nil in `execution.NewEngine()`.

Reply:
removed